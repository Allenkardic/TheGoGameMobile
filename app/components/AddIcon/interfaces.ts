import {GestureResponderEvent, StyleProp, ViewStyle} from 'react-native';

export default interface AddIconProps {
  onPress?: (event: GestureResponderEvent) => void;
  style?: StyleProp<ViewStyle>;
}
