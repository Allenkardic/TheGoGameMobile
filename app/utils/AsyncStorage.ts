import AsyncStorage from '@react-native-async-storage/async-storage';

// Function to set an item in AsyncStorage
export const asyncStore = async (key: string, value: string): Promise<void> => {
  try {
    await AsyncStorage.setItem(key, value);
  } catch (error) {
    console.error(`Error setting item ${key}:`, error);
    throw error;
  }
};

// Function to update an item in AsyncStorage
export const asyncUpdate = async (
  key: string,
  value: string,
): Promise<void> => {
  try {
    await AsyncStorage.mergeItem(key, value);
  } catch (error) {
    console.error(`Error updating item ${key}:`, error);
    throw error;
  }
};

// Function to get an item from AsyncStorage
export const asyncGet = async (key: string): Promise<string | null> => {
  try {
    const value = await AsyncStorage.getItem(key);
    return value;
  } catch (error) {
    console.error(`Error getting item ${key}:`, error);
    throw error;
  }
};

// Function to remove an item from AsyncStorage
export const asyncRemove = async (key: string): Promise<void> => {
  try {
    await AsyncStorage.removeItem(key);
  } catch (error) {
    console.error(`Error removing item ${key}:`, error);
    throw error;
  }
};
