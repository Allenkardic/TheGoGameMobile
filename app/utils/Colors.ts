const tintColor = '#031a61';

const secondVariant = '#4250f2';
// const thirdVariant = '#727cf5';
const thirdVariant = '#0C0932';

const PRIMARY_COLOR = thirdVariant;
const SECONDARY_COLOR = tintColor;

const WHITE = '#FFFFFF';
//  alternative grey - #0000000D

const BLACK = '#36454f';

const GREY = '#6D6E71';
const LIGHT_GREY = '#fffcfc';

const RED = '#E80808'; // red

const GREEN = '#139E5C';

const GRAYALTERNATE = '#AEAEB6';

export default {
  background: '#ffffff',
  tintColor,
  white: WHITE,
  tabIconSelected: PRIMARY_COLOR,
  tabIconDefault: GREY,
  primary: PRIMARY_COLOR,
  secondary: SECONDARY_COLOR,
  tabBar: '#fefefe',
  errorBackground: RED,
  errorText: RED,
  text: '#010a1a',
  altText: '#999999',
  infoText: '#2a3b47',
  muted: '#d5d8d9',
  lightText: GREY,
  darkText: BLACK,
  danger: RED,
  success: '#21951e',
  warningBackground: 'goldenrod',
  warningText: '#666804',
  noticeBackground: tintColor,
  noticeText: '#fff',
  grey: GREY,
  lightGrey: LIGHT_GREY,
  greyAlternate: GRAYALTERNATE,
  greyVariantTwo: '#E2E2E2',
  greyVariantThree: '#AEAEB6',
  greyVariantFour: '#9D9D9D',
  black: BLACK,
  ghostWhite: '#F8F9FF',
  alternateBlack: '#1b1b1b',
  defaultBlack: '#121212',
  gold: '#F79609',
  green: GREEN,
  inputBorderVarientOne: 'rgba(12, 9, 50, 0.14)',
  primaryVariantOne: 'rgba(255, 255, 255, 0.2)',
  placeholder: '#BDBDBD',
  inputLabelText: '#1B1C20',
};

// #F8F9FF
