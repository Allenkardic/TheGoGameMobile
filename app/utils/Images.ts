export default {
  emptyList: require('../../assets/images/list.png'),
};
