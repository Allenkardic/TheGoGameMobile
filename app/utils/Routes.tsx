import {RootStackParamList} from '../navigations/interfaces';

const Routes: {stack: {[K in keyof RootStackParamList]: K}} = {
  stack: {
    HOME: 'HOME',
    ADDTODO: 'ADDTODO',
    UPDATETODO: 'UPDATETODO',
    ONBOARDING: 'ONBOARDING',
  },
};

export default Routes;
