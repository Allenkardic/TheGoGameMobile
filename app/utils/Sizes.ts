import {Dimensions} from 'react-native';

const IS_SMALL_SCREEN = Dimensions.get('window').width <= 320;

const BUTTON_SIZE = IS_SMALL_SCREEN ? 48 : 60;

export default {
  DEFAULT_CARD_BOTTOM_MARGIN: 32,
  INPUT_BORDER_RADIUS: 12,
  SEARCH_INPUT_HEIGHT: 44,
  INPUT_HEIGHT: 50,
  INPUT_BORDER_WIDTH: 1.5,
  INPUT_FONT_SIZE: 14,
  INPUT_LINE_HEIGHT: 18,
  BUTTON_SIZE: IS_SMALL_SCREEN ? 48 : 60,
  BUTTON_BORDER_RADIUS: BUTTON_SIZE * 0.5,
  BUTTON_FONT_SIZE: 16,
  BUTTON_LINE_HEIGHT: 24,
};
