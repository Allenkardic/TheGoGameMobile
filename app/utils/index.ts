export {default as Routes} from './Routes';
export {default as Colors} from './Colors';
export {default as Sizes} from './Sizes';
export {default as Spacing} from './Spacing';
export {default as Images} from './Images';
export {asyncUpdate, asyncGet, asyncStore, asyncRemove} from './AsyncStorage';
