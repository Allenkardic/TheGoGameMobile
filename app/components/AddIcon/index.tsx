import React, {memo} from 'react';
import Icon from '@expo/vector-icons/Feather';
import {Pressable, StyleSheet, Platform} from 'react-native';
import {Colors} from '../../utils';
import AddIconProps from './interfaces';

function AddIcon({onPress, style}: AddIconProps) {
  return (
    <Pressable
      testID="add-icon"
      style={[styles.container, style]}
      onPress={onPress}>
      <Icon name="plus" size={25} color={Colors.white} />
    </Pressable>
  );
}

const styles = StyleSheet.create({
  container: {
    width: 50,
    height: 50,
    borderRadius: 50 / 2,
    backgroundColor: Colors.success,
    alignItems: 'center',
    justifyContent: 'center',
    elevation: 5,
    shadowColor:
      Platform.OS === 'android' ? 'grey' : 'rgba(236, 224, 248, 0.5)',
    shadowOffset: {
      width: 2,
      height: 2,
    },
    shadowOpacity: 1,
    shadowRadius: 3.84,
  },
});

export default memo(AddIcon);
