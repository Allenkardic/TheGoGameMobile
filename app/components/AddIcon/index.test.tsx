import React from 'react';
import {Pressable} from 'react-native';
import {render} from '@testing-library/react-native';
import AddIcon from '.';
import {Colors} from '../../utils';
import renderer from 'react-test-renderer';

describe('AddIcon Component', () => {
  beforeEach(() => {
    jest.useFakeTimers(); // Mock all timers
  });

  afterEach(() => {
    jest.clearAllTimers(); // Clear pending timers
    jest.useRealTimers(); // Reset timers to default behavior
  });

  it('renders correctly', () => {
    const tree = renderer.create(<AddIcon onPress={jest.fn()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onPress when pressed', () => {
    const onPressMock = jest.fn();
    const component = renderer.create(<AddIcon onPress={onPressMock} />);
    const pressable = component.root.findByProps({testID: 'add-icon'});

    // Simulate press event
    pressable.props.onPress();

    // Check if the onPress function was called
    expect(onPressMock).toHaveBeenCalled();
  });

  it('applies styles correctly', () => {
    const {getByTestId} = render(
      <AddIcon onPress={() => {}} style={{backgroundColor: 'red'}} />,
    );

    // Check if the custom styles are applied
    const icon = getByTestId('add-icon');
    expect(icon.props.style[1].backgroundColor).toBe('red');
    expect(icon.props.style[0].backgroundColor).toBe(Colors.success);

    jest.runAllTimers();
  });
});
