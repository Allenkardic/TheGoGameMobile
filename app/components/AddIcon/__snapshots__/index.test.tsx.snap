// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddIcon Component renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#21951e",
        "borderRadius": 25,
        "elevation": 5,
        "height": 50,
        "justifyContent": "center",
        "shadowColor": "rgba(236, 224, 248, 0.5)",
        "shadowOffset": {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 1,
        "shadowRadius": 3.84,
        "width": 50,
      },
      undefined,
    ]
  }
  testID="add-icon"
>
  <Icon
    color="#FFFFFF"
    name="plus"
    size={25}
  />
</View>
`;
